core/src/worker/mod.rs

```diff
@@ -539,23 +581,25 @@ fn build_wf_basics(
 mod tests {
     use super::*;
     use crate::{test_help::test_worker_cfg, worker::client::mocks::mock_workflow_client};
-    use temporal_sdk_core_protos::temporal::api::workflowservice::v1::PollActivityTaskQueueResponse;
-
-    #[tokio::test]
-    async fn activity_timeouts_dont_eat_permits() {
-        let mut mock_client = mock_workflow_client();
-        mock_client
-            .expect_poll_activity_task()
-            .returning(|_, _| Ok(PollActivityTaskQueueResponse::default()));
 
-        let cfg = test_worker_cfg()
-            .max_outstanding_activities(5_usize)
-            .build()
-            .unwrap();
-        let worker = Worker::new_test(cfg, mock_client);
-        assert_eq!(worker.activity_poll().await.unwrap(), None);
-        assert_eq!(worker.at_task_mgr.unwrap().remaining_activity_capacity(), 5);
-    }
+    // TODO: this test is stuck now that the activity poller loops until a task is available
```

Comment:
I commented this out a while ago and the code has changed significantly since then.
I'll check this again after the recent changes.

Reply:
Checked, we'd have to revisit this test.

Reply:
I'm leaving this TODO open

Reply:
I think we can fix this pretty easily by just making sure the mock gets called 1 or more times (which is as easy as adding `.times(1..)`) and then also setting a timeout on the poll. 

So, poll for a bit, see mock returned default, abort polling, assert slots are unchanged.

Reply:
Would still like to fix this before merge

Reply:
Hmm.. but timeouts do eat up permits now.
We don't return anymore, we keep polling in a loop until we get a non-timeout response.
I can change the test to verify that we keep the permit alive as long as the poller times out.

Reply:
Changed.